### PR TITLE
Created Film DTOs, CreateFilm, ReadFilm, DeleteFilm, ListFilm

### PR DESCRIPTION
**Pull request by punamasawa777**
Feature Film Ticket No : #388 
--> Created DTOs
--> Create Film
--> Read Film
--> Delete Film
--> Listing Films
--> Testing e2e

### CODE REVIEW

File: src/components/product/film/film.service.ts [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    ];
+  };
+
+  // helper method for defining properties
+  permission = (
+    property: string,
+    sg: string,
+    baseNode: string,
+    read: boolean,
+    edit: boolean
```
Comment (ibohalsinghkoijam) [inline src/components/product/film/film.service.ts:87]: Please add admin flag in permission node

File: src/components/product/film/film.service.ts [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+          ...this.property('rangeEnd', input.range?.rangeEnd, 'range'),
+          [
+            node('adminSG', 'SecurityGroup', {
+              active: true,
+              createdAt,
+              name: input.name + ' admin',
+            }),
+            relation('out', '', 'member', { active: true, createdAt }),
+            node('requestingUser'),
+          ],
```
Comment (ibohalsinghkoijam) [inline src/components/product/film/film.service.ts:180]: Please refactor the code that you can call  the function `..this.permission` in single for `adminSG` and `readerSG`. 
So, change in helper function `permission` to be able for single call.

File: src/components/product/film/film.service.ts [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+  // async update(input: UpdateFilm, session: ISession): Promise<Film> {
+  //   const film = await this.readOne(input.id, session);
+  //   return this.db.sgUpdateProperties({
+  //     session,
+  //     object: film,
+  //     props: ['name'],
+  //     changes: input,
+  //     nodevar: 'film',
+  //   });
+  // }
```
Comment (candleshine) [inline src/components/product/film/film.service.ts:279]: add `if (input.range) { this.rangeService.update(input.range, session)`

File: src/components/product/film/dto/range.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { Type } from '@nestjs/common';
```
Comment (candleshine) [inline src/components/product/film/dto/range.ts:1]: move this file out of film into product/range

**Pull request merged**